packages/components/src/utils/hooks/use-cx.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { Context } from 'react';
+import { CacheProvider, EmotionCache } from '@emotion/react';
+import type { SerializedStyles } from '@emotion/serialize';
+import { insertStyles } from '@emotion/utils';
+// eslint-disable-next-line no-restricted-imports
+import { cx as innerCx, ClassNamesArg } from '@emotion/css';
+
+/**
+ * WordPress dependencies
+ */
+import { useContext, useCallback } from '@wordpress/element';
+
+// @ts-ignore Private property
+const EmotionCacheContext: Context< EmotionCache > = CacheProvider._context;
+
+const useEmotionCacheContext = () => useContext( EmotionCacheContext );
+
+const isSerializedStyles = ( o: any ): o is SerializedStyles =>
+	[ 'name', 'styles' ].every( ( p ) => typeof o[ p ] !== 'undefined' );
+
+export const useCx = () => {
+	const cache = useEmotionCacheContext();
+
+	const cx = useCallback(
+		( ...classNames: ( ClassNamesArg | SerializedStyles )[] ) => {
+			return innerCx(
+				...classNames.map( ( arg ) => {
+					if ( isSerializedStyles( arg ) ) {
+						insertStyles( cache, arg, false );
+						return `${ cache.key }-${ arg.name }`;
```

Comment:
Is there any chance that this interpolation technique may change in emotion and potentially break this hook?

Reply:
I doubt it, but we'd notice it quickly I would think. Testing that iframes are working with Emotion should be something we do every Emotion upgrade.

Reply:
> Testing that iframes are working with Emotion

This is actually a good point. Should we also add a unit test and/or storybook story to showcase iframe compatibility?

Reply:
Oh yeah, that'd be good to do, I'll add it.